app/src/main/java/android/support/v17/leanback/supportleanbackshowcase/app/room/db/DatabaseHelper.java

```diff
@@ -0,0 +1,179 @@
+
+    private AppDatabase mDb;
+
+    /**
+     * use atomic boolean for synchronization
+     */
+    private final AtomicBoolean mInitialized = new AtomicBoolean(false);
+
+
+    public synchronized static DatabaseHelper getInstance() {
+        if (sInstance == null) {
+            sInstance = new DatabaseHelper();
+        }
+        return sInstance;
+    }
+
+    public LiveData<Boolean> isDatabaseCreated() {
+        return mIsDatabaseCreated;
+    }
+
+    public LiveData<Boolean> getDatabaseUpdatedSignal() {
+        return mDatabaseUpdatedSignal;
+    }
+
+
+    @Nullable
+    public AppDatabase getDatabase() {
+        return mDb;
+    }
+
+    public void createDb(Context context) {
+
+        /**
+         * When we create a new view model, it will try to create a new database. This is why
+         * we use compareAndSet for synchronization
+         */
+        if (!mInitialized.compareAndSet(false, true)) {
+            // Already initializing
+            return;
+        }
+
+        // Trigger an update to show a loading screen.
+        // It can be used as an independent signal to represent the beginning for fetching data
+        // from internet
+        mIsDatabaseCreated.setValue(false);
+
+        new AsyncTask<Context, Void, Boolean>() {
+
+            @Override
+            protected Boolean doInBackground(Context... params) {
+
+                Context context = params[0].getApplicationContext();
+
+                AppDatabase db = Room.databaseBuilder(context.getApplicationContext(),
+                        AppDatabase.class, DATABASE_NAME).build();
+
+                /**
+                 * If database existed,
+                 */
+                if (!context.getDatabasePath(DATABASE_NAME).exists()) {
+                    try {
+                        String url =
+                                "https://storage.googleapis.com/android-tv/android_tv_videos_new.json";
+                        DatabaseInitUtil init = new DatabaseInitUtil();
+                        init.initializeDb(db, url);
+                    } catch (IOException e) {
+                        e.printStackTrace();
+                        return false;
+                    }
+                }
+
+                mDb = db;
+                return true;
+            }
+
+            @Override
+            protected void onPostExecute(Boolean res) {
+                /**
+                 * If database is created unsuccessfully, database will be reinitialized
+                 * next time
+                 */
+                if (!res) {
+                    //
+                    // make sure there is no database crash happen
+                    //
+                    if (mInitialized.compareAndSet(true, false)) {
+                        mIsDatabaseCreated.setValue(res);
+                    }
+                }
+                mIsDatabaseCreated.setValue(res);
```

Comment:
see comment above.
also, LiveData holds data, it is not suitable for dispatching events (which seems like you are dispatching events here)

Reply:
The mIsDatabaseCreated is just used as a signal to notify other live data that the database is ready to access.